march_gazebo_plugins/src/obstacle_controller.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 Project March.
+
+#include <obstacle_controller.h>
+
+namespace gazebo
+{
+ObstacleController::ObstacleController(physics::ModelPtr model) : model_(model)
+{
+  this->foot_left_ = this->model_->GetLink("ankle_plate_left");
+  this->foot_right_ = this->model_->GetLink("ankle_plate_right");
+
+  this->subgait_name_ = "home_stand";
+  this->subgait_changed_ = true;
+}
+
+void ObstacleController::newSubgait(const march_shared_resources::GaitActionGoalConstPtr& _msg)
+{
+  if (this->subgait_name_ == "right_open" or this->subgait_name_ == "right_swing" or
+      this->subgait_name_ == "left_swing")
+  {
+    this->swing_step_size_ = 0.8 * this->swing_step_size_ + 0.4 * std::abs(this->foot_right_->WorldPose().Pos().X() -
```

Comment:
Where do the 0.8 and 0.4 come from. Are these the ideal values to optimize the step size estimation?

Reply:
Added a comment